### PR TITLE
Fix Portal Fill Issues

### LINKED ISSUE #549: Filling a Portal with Flint and Steel on Z Axis
Filling a Portal with Flint and Steel on the Z Axis causes the nether_portal blocks to be placed on the x-axis, and therefore appear incorrectly.

Currently a workaround is to use the Vanilla `fill` command.

### PR DESCRIPTION
**Pull request by nicegamer7**
This PR fixes the orientation of the nether_portal blocks placed along the z-axis when filling a portal using flint and steel. This closes #549.

### CODE REVIEW

**Comment by nicegamer7** [comment]
I've just added two commits:
- The first makes sure to use physics on all fill materials except the `nether_portal` block. The reason for this is that Spigot says in their Javadocs that setting it to false is generally unsafe. The only reason I've set it to false for `nether_portal` blocks is because after the portal is filled, if the blocks are broken, the console will send an error like this:
```[Server thread/ERROR]: POI data mismatch: never registered at BlockPosition{x=1614, y=64, z=-217}```
- The second fixes filling portals with air via an empty bucket. The issue was that the clicked block location was being translated away from the portal, towards the user. But because PlayerBucketFillEvent returns the block that filled the bucket, there was no need to translate the location.

I've tested all these changes on 1.13.2, 1.14.4, and 1.15.2, and they are all working as expected. Lava/Water do not flow, `nether_portal` blocks are rotated if needed, and portals are filled with air when an empty bucket is filled.

File: src/main/java/com/onarandombox/MultiversePortals/listeners/MVPPlayerListener.java [hunk 1]
```diff
@@ -62,21 +62,21 @@ public void playerTeleport(PlayerTeleportEvent event) {
 
     @EventHandler(priority = EventPriority.LOW)
     public void playerBucketFill(PlayerBucketFillEvent event) {
-        Location translatedLocation = this.getTranslatedLocation(event.getBlockClicked(), event.getBlockFace());
         this.plugin.log(Level.FINER, "Fill: ");
         this.plugin.log(Level.FINER, "Block Clicked: " + event.getBlockClicked() + ":" + event.getBlockClicked().getType());
-        this.plugin.log(Level.FINER, "Translated Block: " + event.getPlayer().getWorld().getBlockAt(translatedLocation) + ":" + event.getPlayer().getWorld().getBlockAt(translatedLocation).getType());
+        // PlayerBucketFillEvent.getBlockClicked() returns the block that filled the bucket, so no need to translate the location!
+        this.plugin.log(Level.FINER, "No need to translate this location!");
```
Comment (dumptruckman) [inline src/main/java/com/onarandombox/MultiversePortals/listeners/MVPPlayerListener.java:68]: If we're not translating it anywhere I'm not sure we need this log message.

File: src/main/java/com/onarandombox/MultiversePortals/utils/PortalFiller.java [hunk 2]
```diff
@@ -58,7 +60,17 @@ public boolean fillRegion(MultiverseRegion r, Location l, Material type) {
...
+            if (type == Material.NETHER_PORTAL) {
+                // for now we won't use physics because it causes errors
+                newLoc.setType(type, false);
+                if (useX == 0) {
+                    Orientable b = (Orientable) newLoc.getBlockData();
+                    b.setAxis(Axis.Z);
+                    // also don't use physics here
+                    newLoc.setBlockData(b, false);
+                }
+            } else newLoc.setType(type);
```
Comment (dumptruckman) [inline src/main/java/com/onarandombox/MultiversePortals/utils/PortalFiller.java:73]: please use
```java
else {
    newLoc.setType(type);
}
```
This is mostly just a general style nit pick but I would argue that this shortened style can lead to issues for future developers (forgetting to put in braces when adding an extra line and such). The only time I tend to think the short style is okay is
```java
if (something) return;
```

File: src/main/java/com/onarandombox/MultiversePortals/listeners/MVPPlayerListener.java (hunk 1 above)
Comment (nicegamer7) [reply src/main/java/com/onarandombox/MultiversePortals/listeners/MVPPlayerListener.java:68]: Good point.

File: src/main/java/com/onarandombox/MultiversePortals/utils/PortalFiller.java (hunk 2 above)
Comment (nicegamer7) [reply src/main/java/com/onarandombox/MultiversePortals/utils/PortalFiller.java:73]: Sure, I'll amend this today and keep it in mind for future PRs.

File: src/main/java/com/onarandombox/MultiversePortals/listeners/MVPPlayerListener.java [hunk 3]
```diff
@@ -62,21 +62,20 @@ public void playerTeleport(PlayerTeleportEvent event) {
 
     @EventHandler(priority = EventPriority.LOW)
     public void playerBucketFill(PlayerBucketFillEvent event) {
-        Location translatedLocation = this.getTranslatedLocation(event.getBlockClicked(), event.getBlockFace());
         this.plugin.log(Level.FINER, "Fill: ");
         this.plugin.log(Level.FINER, "Block Clicked: " + event.getBlockClicked() + ":" + event.getBlockClicked().getType());
-        this.plugin.log(Level.FINER, "Translated Block: " + event.getPlayer().getWorld().getBlockAt(translatedLocation) + ":" + event.getPlayer().getWorld().getBlockAt(translatedLocation).getType());
+        // PlayerBucketFillEvent.getBlockClicked() returns the block that filled the bucket, so no need to translate the location!
```
Comment (dumptruckman) [inline src/main/java/com/onarandombox/MultiversePortals/listeners/MVPPlayerListener.java:67]: This comment is going to not make any sense without the code that is being removed. :)

**Review by dumptruckman — CHANGES_REQUESTED** [review]

File: src/main/java/com/onarandombox/MultiversePortals/listeners/MVPPlayerListener.java (hunk 3 above)
Comment (nicegamer7) [reply src/main/java/com/onarandombox/MultiversePortals/listeners/MVPPlayerListener.java]: Got rid of it 👍.

**Review by dumptruckman — APPROVED** [review]
sweet

**Pull request merged**